lib/rules/yoda.js

```diff
@@ -219,46 +221,56 @@ module.exports = {
+            const textBeforeOperator = sourceCode.getText().slice(node.left.range[1], operatorToken.range[0]);
+            const textAfterOperator = sourceCode.getText().slice(operatorToken.range[1], node.right.range[0]);
+            const leftText = sourceCode.getText(node.left);
+            const rightText = sourceCode.getText(node.right);
+
+            return rightText + textBeforeOperator + OPERATOR_FLIP_MAP[operatorToken.value] + textAfterOperator + leftText;
+        }
+
         //--------------------------------------------------------------------------
         // Public
         //--------------------------------------------------------------------------
 
         return {
-            BinaryExpression: always ? function(node) {
-
-                // Comparisons must always be yoda-style: if ("blue" === color)
-                if (
-                    (node.right.type === "Literal" || looksLikeLiteral(node.right)) &&
-                    !(node.left.type === "Literal" || looksLikeLiteral(node.left)) &&
-                    !(!isEqualityOperator(node.operator) && onlyEquality) &&
-                    isComparisonOperator(node.operator) &&
-                    !(exceptRange && isRangeTest(context.getAncestors().pop()))
-                ) {
-                    context.report({
-                        node,
-                        message: "Expected literal to be on the left side of {{operator}}.",
-                        data: {
-                            operator: node.operator
-                        }
-                    });
-                }
-
-            } : function(node) {
+            BinaryExpression(node) {
+                const expectedLiteral = always ? node.left : node.right;
+                const expectedNonLiteral = always ? node.right : node.left;
 
-                // Comparisons must never be yoda-style (default)
+                // If `expectedLiteral` is not a literal, and `expectedNonLiteral` is a literal, raise an error.
                 if (
-                    (node.left.type === "Literal" || looksLikeLiteral(node.left)) &&
-                    !(node.right.type === "Literal" || looksLikeLiteral(node.right)) &&
+                    (expectedNonLiteral.type === "Literal" || looksLikeLiteral(always ? node.right : node.left)) &&
+                    !(expectedLiteral.type === "Literal" || looksLikeLiteral(always ? node.left : node.right)) &&
                     !(!isEqualityOperator(node.operator) && onlyEquality) &&
                     isComparisonOperator(node.operator) &&
                     !(exceptRange && isRangeTest(context.getAncestors().pop()))
                 ) {
                     context.report({
                         node,
-                        message: "Expected literal to be on the right side of {{operator}}.",
+                        message: `Expected literal to be on the ${always ? "left" : "right"} side of {{operator}}.`,
```

Comment:
In order to facilitate possible i18n someday, could the "left"/"right" interpolation be performed using our double-brace substitution (i.e., same as how "{{operator}}" is handled)? Thanks!